### ISSUE TITLE
Brother HL-L2320D not printing image based print jobs. It flashes processing and returns to a steady green ready light.

### ISSUE THREAD

**Issue by adam-jones-net** [OpenPrinting/cups#680]
As the title says, I can print text documents to my HL-L2320D, but not images.    The printer always starts flashing its green ready light for some seconds before returning to a constant green ready light.

The queue is empty, the job appears as completed and cups is ready for more jobs; it doesn't crash or stick at the point of this first job failure.

I've had the debug option enabled in cups config and there are no error remarks of note.   I do see "HTTP_STATE_WAITING Closing for error 32 (Broken pipe)" in the log sometimes although it isn't marked as an error.

I'm using the Brother drivers from https://github.com/pdewacht/brlaser  with default options.   I've seen comments about printing at 300dpi which fixed similar issues for some users, but that isn't an option for me as the printer doesn't print at lower than 600dpi, which it defaults to.

I've also run through all the steps that seem relevant [here](https://wiki.archlinux.org/title/CUPS/Troubleshooting) to no help.

Some output from the CLI if useful, can anyone assist me troubleshoot this ?

**lpinfo -v**
file cups-brf:/
network beh
network ipp
network https
network ipps
network socket
network http
direct usb://Brother/HL-L2320D%20series?serial=U63877C2N833460

**lpstat -t**
scheduler is running
system default destination: Brother_HL-L2320D_series
device for Brother_HL-L2320D_series: usb://Brother/HL-L2320D%20series?serial=U63877C2N833460
Brother_HL-L2320D_series accepting requests since Sun 30 Apr 2023 13:18:19 -04
printer Brother_HL-L2320D_series is idle.  enabled since Sun 30 Apr 2023 13:18:19 -04

I'm running CUPS CUPS 2.3.3op2 on Raspbian GNU/Linux 11 (bullseye)

**Comment by zdohnal** [comment]
Hi,

can you try reproducing it on the latest upstream version - 2.4.4?

Do let me know once you have upgraded.

**Comment by zdohnal** [comment]
Then it would be great if you reported this to Debian. We try to provide only best effort support for older releases. I can check debug logs if you provide some (run ```cupsctl LogLevel=debug2```, try to print the image, and then get /var/log/cups/error_log file and copy it into file error_log.txt, which you can attach here as a file), or I recommend running filters which are run before ```rastertobrlaser``` and review their output.

You can run filters by hand like this:

```
sudo PPD=/etc/cups/ppd/<printer name>.ppd /usr/lib/cups/filter/<filter name> 1 <user> '' 1 '' <file> > out
```

where <printer name> is print queue/printer name of your printer in CUPS, <filter name> is name of filter (you can see which filters are run from debug logging), <user> your username and <file> is filename/path to the file you want to print. If everything goes well, you can inspect ```out``` file and check its mime type and contents. ```file --mime out``` will show you mime type and reviewing tool depends on the file type - you can check it in text editor for types you don't have viewer for.

In your case it will be a raster, which you can open by ```rasterview``` command (at least that's what I got when I used driver for your printer and printed into a file). I'm not sure whether it is available on Debian though.

**Comment by adam-jones-net** [comment]
I've managed to fix by installing 2.4.x on my Pi.  This was done by actually reinstalling the system and using Armbian instead of the defecto Pi OS.   Armbian supports newer repositories out of the box, that at the time of writing this let me install 2.4.1 instead of 2.3.3, which was the latest available in the packages on Pi OS.
